File: app/src/main/java/com/mirash/passkeeper/activity/edit/CredentialsItemCallback.java
package com.mirash.passkeeper.activity.edit;

import com.mirash.passkeeper.motion.OnStartDragListener;
import com.mirash.passkeeper.model.CredentialsItem;

import java.util.List;

/**
 * @author Mirash
 */
public interface CredentialsItemCallback extends OnStartDragListener {
    void onLinkClick(String link);

    void onEditClick(CredentialsItem item);

    void onOrderChanged(List<CredentialsItem> items);

    void onShare(CredentialsItem item);
}

File: app/src/main/java/com/mirash/familiar/activity/main/credentials/CredentialsItemCallback.kt
package com.mirash.familiar.activity.main.credentials

import com.mirash.familiar.model.credentials.ICredentials
import com.mirash.familiar.motion.OnStartDragListener

/**
 * @author Mirash
 */
interface CredentialsItemCallback : OnStartDragListener {
    fun onLinkClick(link: String)
    fun onEditClick(item: ICredentials)
    fun onOrderChanged(items: List<ICredentials>)
}
